subsys/net/lib/sockets/sockets.c

```diff
@@ -2825,3 +2950,158 @@ static bool inet_is_supported(int family, int type, int proto)
 NET_SOCKET_REGISTER(af_inet46, NET_SOCKET_DEFAULT_PRIO, AF_UNSPEC,
 		    inet_is_supported, zsock_socket_internal);
 #endif /* CONFIG_NET_NATIVE */
+
+
+#if defined(CONFIG_NET_SOCKETS_OBJ_CORE)
```

Comment:
I wonder, maybe this new stat-related functions/resources could be moved to a separate, new file? `sockets.c` is already quite loaded.